### PR TITLE
[null-safety] update tests and tool auto-detection for null safe dart

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes auto-detection of sound mode in web projects

### CODE REVIEW

**Review by zanderso — APPROVED** [review]
lgtm w/o the roll

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 1]
```diff
@@ -94,16 +95,35 @@ class FlutterDevice {
...
-          ...?buildInfo?.extraFrontEndOptions,
-          if (!(buildInfo?.extraFrontEndOptions?.contains('--sound-null-safety') ?? false))
-            '--sound-null-safety'
-        ];
+        extraFrontEndOptions =  buildInfo.extraFrontEndOptions;
+      } else {
+        final LanguageVersion languageVersion = target != null
+          ? packageConfig?.packageOf(Uri.parse(target))?.languageVersion
+          : null;
+        final bool isNullSafe = languageVersion != null && languageVersion.major >= 2 && languageVersion.minor >= 11;
```
Comment (jakemac53) [inline packages/flutter_tools/lib/src/resident_runner.dart:113]: A null language version is supposed to mean opt in, fwiw

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart:113]: TIL!

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Thanks - updated to be null safe if the language version is null or is gte 2.11

Comment (jakemac53) [reply packages/flutter_tools/lib/src/resident_runner.dart]: How are we going to handle the 2.12 flip here? That will be the sdk where its actually turned on

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: I'll file an issue to adjust the constraints. This is only an approximation of the autodetect logic, so its going to have some holes.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: https://github.com/flutter/flutter/issues/69416

**Pull request merged**